docs-src/dev-guide/sdk-features.md

```diff
@@ -0,0 +1,571 @@
+
+### Signals
+
+A Signal is an asynchronous request to a [Workflow Execution](/workflows#workflow-execution).
+
+### Queries
+
+A Query is a synchronous operation that is used to get the state of a [Workflow Execution](/workflows#workflow-execution).
+The state of a running Workflow Execution is constantly changing.
+You can use Queries to expose the internal Workflow Execution state to the external world.
+Queries are available for running or completed Workflows Executions only if the Worker is up and listening on the Task Queue.
+
+### Workflow Update
+
+An Workflow Update is a request to and a response from a Temporal Client to a Workflow Execution.
+
+The Workflow must have a function to handle the Update.
+Unlike a Signal handler, the Update handler function can mutate the state of the Workflow while also returning a value to the caller.
+The Update handler listens for Updates by the Update's name.
+
+When there is the potential for multiple Updates to cause a duplication problem, Temporal recommends adding idempotency logic to your Update handler that checks for duplicates.
+
+For more information, see [What is an Update?](/workflows#update).
+
+## Remote codec
+
+| Feature                       | Go  | Java | TypeScript | Python | .Net | PHP |
+| ----------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Remote codec](#remote-codec) | Yes | Yes  | Yes        | Yes    | Yes  | No  |
+
+The remote codec refers to exposing a Payload Codec through HTTP endpoints for remote encoding and decoding.
+It can be configured using a remote codec server, where each developer can control the mapping locally through the hosts file.
+However, distributing encryption keys for decoding payloads poses a security risk.
+The authorization header can be set on requests to the codec server.
+
+## Replayer
+
+| Feature               | Go  | Java | TypeScript | Python | .Net | PHP |
+| --------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Replayer](#replayer) | Yes | Yes  | Yes        | Yes    | Yes  | No  |
+
+A Replay is the method by which a Workflow Execution resumes making progress.
+During a Replay the Commands that are generated are checked against an existing Event History.
+Replays are necessary and often happen to give the effect that Workflow Executions are resumable, reliable, and durable.
+
+For more information, see [Deterministic constraints](#deterministic-constraints).
+
+If a failure occurs, the Workflow Execution picks up where the last recorded event occurred in the Event History.
+
+## Retries
+
+| Feature             | Go  | Java | TypeScript | Python | .Net | PHP |
+| ------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Retries](#retries) | Yes | Yes  | Yes        | Yes    | Yes  | Yes |
+
+A Retry Policy is a collection of attributes that instructs the Temporal Server how to retry a failure of a Workflow Execution or an Activity Task Execution.
+
+## SAGA
```

Comment:
Saga is a pattern not a feature, and all SDKs support it (even if one or two SDKs made little inflexible helpers)